storage/imxrt-flash/flashsrv.c

```diff
@@ -12,78 +12,290 @@
+{
+	flash_memory_t *flash_memory = flashsrv_common.flash_memories + fID;
+
+	if (flash_memory->fStatus == flash_memory_unactive)
+		return -EINVAL;
+
+	return flash_bufferedPagesWrite(&flash_memory->ctx, offset, data, size);
+}
+
+
+static int flashsrv_read(uint8_t fID, size_t offset, char *data, size_t size)
+{
+	flash_memory_t *flash_memory = &flashsrv_common.flash_memories[fID];
+
+	if (flash_memory->fStatus == flash_memory_unactive)
+		return -EINVAL;
+
+	return flash_readData(&flash_memory->ctx, offset, data, size);
+}
+
+
+static int flashsrv_eraseSector(unsigned char fID, unsigned int offs)
+{
+	flash_memory_t *flash_memory = flashsrv_common.flash_memories + fID;
+
+	if (flash_memory->fStatus == flash_memory_unactive)
+		return -EINVAL;
+
+	return flash_sectorErase(&flash_memory->ctx, offs);
+}
+
+
+static int flashsrv_chipErase(unsigned char fID)
+{
+	flash_memory_t *flash_memory = flashsrv_common.flash_memories + fID;
+
+	if (flash_memory->fStatus == flash_memory_unactive)
+		return -EINVAL;
+
+	return flash_chipErase(&flash_memory->ctx);
+}
+
+
+/* Callbacks to meterfs - wrappers for basic flash functions */
+
+static size_t flashsrv_mfsWritef0(unsigned int addr, void *buff, size_t bufflen)
+{
+	flash_memory_t *flash_memory = flashsrv_common.flash_memories + 0;
+
+	if (flash_memory->fStatus == flash_memory_unactive)
+		return -EINVAL;
+
+	return flash_directBytesWrite(&flash_memory->ctx, addr, buff, bufflen);
+}
+
+
+static size_t flashsrv_mfsWritef1(unsigned int addr, void *buff, size_t bufflen)
+{
+	flash_memory_t *flash_memory = flashsrv_common.flash_memories + 1;
+
+	if (flash_memory->fStatus == flash_memory_unactive)
+		return -EINVAL;
+
+	return flash_directBytesWrite(&flash_memory->ctx, addr, buff, bufflen);
+}
+
+
+static size_t flashsrv_mfsReadf0(unsigned int addr, void *buff, size_t bufflen)
+{
+	return flashsrv_read(0, addr, buff, bufflen);
+}
+
+
+static size_t flashsrv_mfsReadf1(unsigned int addr, void *buff, size_t bufflen)
+{
+	return flashsrv_read(1, addr, buff, bufflen);
+}
+
+
+static void flashsrv_mfsEraseSectorf0(unsigned int addr)
+{
+	flashsrv_eraseSector(0, addr);
+}
+
+
+static void flashsrv_mfsEraseSectorf1(unsigned int addr)
+{
+	flashsrv_eraseSector(1, addr);
+}
+
+
+static void flashsrv_mfsPartitionErasef0(void)
 {
 	int i;
-	for (i = 0; i < FLASHES_NO; ++i) {
-		if (oid.id == flashsrv_common.flash_memories[i].oid.id)
-			return flashsrv_common.flash_memories + i;
-	}
-	return NULL;
+	uint32_t sectorsNb;
+	flash_memory_t *flash_memory;
+
+	flash_memory = flashsrv_common.flash_memories + 0;
+	sectorsNb = flash_memory->parts[flash_memory->currPart].sz / flash_memory->ctx.properties.sector_size;
+
+	for (i = 0; i < sectorsNb; ++i)
+		flashsrv_eraseSector(0, flash_memory->parts[flash_memory->currPart].offs + flash_memory->ctx.properties.sector_size * i);
 }
 
 
-static void flashsrv_devCtl(msg_t *msg)
+static void flashsrv_mfsPartitionErasef1(void)
```

Comment:
Copy-pasted code, only one parameter to flashsrv_eraseSector is different in comparison with flashsrv_mfsPartitionErasef0